src/libinit.js

```diff
@@ -126,7 +137,8 @@ export const LibraryInitGeneratorMixin = (B) => class extends B {
 	_handlePrompts(data) {
 		Object.assign(this.options, data);
 		if (this.options.name) {
-			this.options.Name = capitalizeFirstLetter(this.options.name);
+			this.options.name_code = lowercaseFirstLetter(this.options.name.replace(/[^a-zA-Z0-9_]/g, ''));
```

Comment:
We decided that the name of the library could have dashes and pluses, but the generated C code identifiers can't. I'm happy with the choice I made here.